File: src/main/java/fi/finlit/edith/ui/services/svn/FileItemWithDocumentId.java
package fi.finlit.edith.ui.services.svn;

import java.util.List;


public class FileItemWithDocumentId extends FileItem {
    private final Long documentId;

    private final boolean isSelected;

    private final long noteCount;

    public FileItemWithDocumentId(
            String title,
            String path,
            boolean isFolder,
            List<FileItem> children,
            boolean hasChildren,
            Long documentId,
            boolean isSelected,
            long noteCount) {
        super(title, path, isFolder, children, hasChildren);
        this.documentId = documentId;
        this.isSelected = isSelected;
        this.noteCount = noteCount;
    }

    public Long getDocumentId() {
        return documentId;
    }

    public boolean isSelected() {
        return isSelected;
    }

    public long getNoteCount() {
        return noteCount;
    }
}

File: src/main/scala/fi/finlit/edith/ui/services/svn/FileItemWithDocumentId.scala
package fi.finlit.edith.ui.services.svn

import java.util.List
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

class FileItemWithDocumentId(title: String, 
    path: String, 
    isFolder: Boolean, 
    children: List[FileItem], 
    hasChildren: Boolean, 
    @BeanProperty val documentId: java.lang.Long, 
    isSelected: Boolean, 
    @BeanProperty val noteCount: Long) extends FileItem(title, path, isFolder, children, hasChildren) {



  def isSelected(): Boolean = isSelected
}
